### PR TITLE
Add repository owner

### PR DESCRIPTION
**Pull request by aymanbagabas**
- Each repository now has an owner id with full access rights to the repository.
- Fix: separate LFS objects on a per-repository bases

### CODE REVIEW

File: server/backend/repo.go [hunk 1]
```diff
@@ -92,37 +105,42 @@ func (d *Backend) ImportRepository(ctx context.Context, name string, remote stri
...
-	if err := git.Clone(remote, rp, copts); err != nil {
+	cmd := exec.CommandContext(ctx, "git", "clone", "--bare", "--mirror", remote, ".")
+	cmd.Env = append(cmd.Env,
+		fmt.Sprintf(`GIT_SSH_COMMAND=ssh -o UserKnownHostsFile="%s" -o StrictHostKeyChecking=no -i "%s"`,
+			filepath.Join(d.cfg.DataPath, "ssh", "known_hosts"),
+			d.cfg.SSH.ClientKeyPath,
+		),
+	)
+	cmd.Dir = rp
+	if err := cmd.Run(); err != nil {
```
Comment (caarlos0) [inline server/backend/repo.go:120]: bug in go-git?

File: server/db/migrate/0004_repo_owner_postgres.down.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+ALTER TABLE repos DROP COLUMN user_id;
```
Comment (caarlos0) [inline server/db/migrate/0004_repo_owner_postgres.down.sql:1]: same comment from the other day: I would leave the down.sql files empty...

File: server/db/models/repo.go [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
-	CreatedAt   time.Time `db:"created_at"`
-	UpdatedAt   time.Time `db:"updated_at"`
+	ID          int64         `db:"id"`
+	Name        string        `db:"name"`
+	ProjectName string        `db:"project_name"`
+	Description string        `db:"description"`
+	Private     bool          `db:"private"`
+	Mirror      bool          `db:"mirror"`
+	Hidden      bool          `db:"hidden"`
+	UserID      sql.NullInt64 `db:"user_id"`
```
Comment (caarlos0) [inline server/db/models/repo.go:17]: could probably be `NullUint64`

**Review by caarlos0 — APPROVED** [review]

File: server/db/migrate/0004_repo_owner_postgres.down.sql (hunk 2 above)
Comment (aymanbagabas) [reply server/db/migrate/0004_repo_owner_postgres.down.sql:1]: if not, rolling back and then migrating again won't succeed :/

Comment (caarlos0) [reply server/db/migrate/0004_repo_owner_postgres.down.sql:1]: you can fix that by using `if [not] exists` in the `up.sql`:

```sql
ALTER TABLE if exists foo add if not exists bar type;
CREATE TABLE if not exists foo;
```

Comment (aymanbagabas) [reply server/db/migrate/0004_repo_owner_postgres.down.sql:1]: Sqlite doesn't support that

Comment (caarlos0) [reply server/db/migrate/0004_repo_owner_postgres.down.sql:1]: ah, can probably do the same with `begin` blocks with `exception` handling

in any case, I don't see why people would update and rollback and then update... whats the use case for that?

**Pull request merged**